src/engine/strat_engine/mdv.rs

```diff
@@ -37,6 +37,51 @@ const FILESYSTEM_DIR: &'static str = "filesystems";
 pub struct MetadataVol {
     dev: LinearDev,
     mount_pt: PathBuf,
+    is_mounted: bool,
+}
+
+/// A helper struct that borrows the MetadataVol and ensures that the MDV is
+/// mounted as long as it is borrowed, and then unmounted.
+#[derive(Debug)]
+struct MountedMDV<'a> {
+    mdv: &'a mut MetadataVol,
+}
+
+impl<'a> MountedMDV<'a> {
+    /// Borrow the MDV and ensure it's mounted.
+    fn mount(mdv: &mut MetadataVol) -> EngineResult<MountedMDV> {
+        if !mdv.is_mounted {
+            match mount(Some(&mdv.dev.devnode()),
+                        &mdv.mount_pt,
+                        Some("xfs"),
+                        MsFlags::empty(),
+                        None as Option<&str>) {
+                Err(nix::Error::Sys(nix::Errno::EBUSY)) => {
+                    // We thought we weren't mounted but we actually were
+                    Ok(())
+                }
+                Err(err) => Err(err),
+                Ok(_) => Ok(()),
+            }?;
+
+            mdv.is_mounted = true;
+        }
+
+        Ok(MountedMDV { mdv })
+    }
+
+    fn mount_pt(&self) -> &Path {
```

Comment:
I'm not sure this method serves any purpose.

Reply:
once MountedMDV is created, the MetadataVol is no longer accessible, but some methods still need access to the mount point path.

Reply:
I think it's conceptually simpler to borrow the entire struct so I'm leaving this as is.